gammapy/maps/wcsnd.py

```diff
@@ -446,87 +464,36 @@ def plot(self, ax=None, fig=None, add_cbar=False, stretch='linear', **kwargs):
-
-        Examples
-        --------
+    def _plot_format_allsky(self, ax):
+        # Remove frame
+        ax.coords.frame.set_linewidth(0)
 
-        You can try this out e.g. using a Fermi-LAT diffuse model cube with an energy axis::
+        # Set plot axis limits
+        ymax, xmax = self.data.shape
+        xmargin, _ = self.geom.coord_to_pix({'lon': 180, 'lat': 0})
+        _, ymargin = self.geom.coord_to_pix({'lon': 0, 'lat': -90})
 
-            %matplotlib inline
-            from gammapy.maps import Map
+        ax.set_xlim(xmargin, xmax - xmargin)
+        ax.set_ylim(ymargin, ymax - ymargin)
 
-            m = Map.read("$GAMMAPY_EXTRA/datasets/vela_region/gll_iem_v05_rev1_cutout.fits")
-            m.plot_interactive(cmap='gnuplot2')
-        """
-        import matplotlib.pyplot as plt
-        from astropy.visualization import simple_norm
-        from ipywidgets.widgets.interaction import interact, fixed
-        import ipywidgets as widgets
+        ax.text(0, ymax, self.geom.coordsys + ' coords')
 
-        kwargs.setdefault('interpolation', 'nearest')
-        kwargs.setdefault('origin', 'lower')
-        kwargs.setdefault('cmap', 'afmhot')
-
-        @interact(
-            index=widgets.IntSlider(min=0, max=self.data.shape[0] - 1, step=1, value=1,
-                                    description=self.geom.axes[0].name + ' slice'),
-            stretch=widgets.RadioButtons(options=['linear', 'sqrt', 'log'], value='sqrt',
-                                         description='Plot stretch'),
-            ax=fixed(ax),
-            fig=fixed(fig),
-        )
-        def _plot_interactive(index, stretch, ax=None, fig=None):
-            if self.geom.is_image:
-                raise TypeError('Use .plot() for 2D Maps')
-
-            if fig is None:
-                fig = plt.gcf()
-
-            if ax is None:
-                ax = fig.add_subplot(1, 1, 1, projection=self.geom.wcs)
-
-            axes = self.geom.axes[0]
-
-            data = self.get_image_by_idx([index]).data
-            norm = simple_norm(data[np.isfinite(data)], stretch)
-
-            caxes = ax.imshow(data, norm=norm, **kwargs)
-            fig.colorbar(caxes, ax=ax)
-            ax.set_title(
-                '{:.2f}-{:.2f} {} '.format(
-                    axes.edges[index], axes.edges[index + 1],
-                    self.geom.axes[0].unit.name,
-                )
-            )
-
-            try:
-                ax.coords['glon'].set_axislabel('Galactic Longitude')
-                ax.coords['glat'].set_axislabel('Galactic Latitude')
-            except KeyError:
-                ax.coords['ra'].set_axislabel('Right Ascension')
-                ax.coords['dec'].set_axislabel('Declination')
-            except AttributeError:
-                log.info("Can't set coordinate axes. No WCS information available.")
-
-            plt.show()
+        # Grid and ticks
+        glon_spacing, glat_spacing = 45, 15
+        try:
```

Comment:
I think you can remove this try-except and use the generic way to get the axis from the tuple:
```
lon, lat = ax.coords
```
http://docs.astropy.org/en/stable/visualization/wcsaxes/ticks_labels_grid.html